lib/archethic/beacon_chain.ex

```diff
@@ -260,64 +262,99 @@ defmodule Archethic.BeaconChain do
   end
 
   @doc """
-  Request from the beacon chains all the summaries for the given dates and aggregate them
-
-  ```
-   [0, 1, ...]    Subsets
-      / | \
-     /  |  \
-  [ ]  [ ]  [ ]  Node election for each dates to sync
-   |\  /|\  /|
-   | \/ | \/ |
-   | /\ | /\ |
-  [ ]  [ ]  [ ] Partition by node
-   |    |    |
-  [ ]  [ ]  [ ] Aggregate addresses
-   |    |    |
-  [ ]  [ ]  [ ] Fetch summaries
-   |    |    |
-  [ ]  [ ]  [ ] Aggregate and consolidate summaries
-   \    |    /
-    \   |   /
-     \  |  /
-      \ | /
-       [ ]
-  ```
+  Request from the beacon chains all the summaries for the given date and aggregate them.
+
+  There are 256 concurrent requests (at worst there might be subsets_count(=256) * storage_nodes_count(=197) requests to do here)
+  Then there is a trick to reduce the summaries while still in a stream to avoid the need to store a list of all summaries in memory.
+
+  (The SummaryAggregate is much smaller than the entire list of summaries because of the mechanism that removes duplicate)
+  FYI: We used to use Flow here, but we noticed a very high memory footprint.
   """
   @spec fetch_and_aggregate_summaries(DateTime.t(), list(Node.t())) :: SummaryAggregate.t()
   def fetch_and_aggregate_summaries(date = %DateTime{}, download_nodes) do
```

Comment:
I think it could be interesting to add a telemetry of the execution time of this function